### PR TITLE
retool service-cluster page

### PR DESCRIPTION
**Pull request by BillMills**
I couldn't follow the Interlock Service Cluster page as-is; this is what I needed to do to get multiple service clusters nominally working in UCP 3.2.3.

Docs team: as always please let me know if this PR is in the wrong place, stylistic problems, any concerns of any kind.

@trapier @rupakg @traci-morrison @arueth this is what I was talking about on slack last week, if you'd ptal. Things to consider:

 - does this work for you?
 - does it accurately capture why you'd want to set up service clusters?
 - is it clear (as possible) and free of unnecessary distractions?

Especially note the surprising need-to-manually-delete-the-old-proxy bit that @arueth and I pointed out. Works every time for me, but are we sure that's what's intended?

### CODE REVIEW

**Comment by rupakg** [comment]
@BillMills I like the exhaustive explanation of the config file but I would like to have the example to be there as well. It would make it clearer for people who go by code. But, the explanation will guide them.

I thought there were two more keys that needs to be added as well:
```

**Comment by BillMills** [comment]
@rupakg we could give a complete example, but I'm worried that'll be cut-and-paste bait which will definitely not work: the UCP cluster ID will be wrong.

Then again, it might be easier to correct that than to get all those individual pieces correct, which I expect an extremely low success rate on.

I never had to add the `Config` keys you mentioned, but maybe I'm wrong and just didn't notice what that breaks - we can certainly add them in.

**Comment by BillMills** [comment]
@rupakg per your slack comments, you're right the config service is missing this way - OK, I'll amend to suit later this week.

**Comment by rupakg** [comment]
@BillMills awesome. thx for your work to iron this out.

**Comment by BillMills** [comment]
@rupakg - now that I include those keys, actually I still don't have a `ucp-interlock-config` service; there must be something else missing, TBD.

**Comment by BillMills** [comment]
@rupakg ah sorry, now that I think about this carefully, we're conflating two things here - the `config` services are for if you're enabling hitless proxy updates, but this article is about enabling service clusters. Those are separate things deserving separate articles.

**Comment by rupakg** [comment]
@BillMills yeah you are right on that. lets keep that out.

**Comment by BillMills** [comment]
OK, I tried this again from the top and it all works for me. Docs team please let me know any thoughts.

File: ee/ucp/interlock/usage/service-clusters.md [hunk 1]
```diff
@@ -4,205 +4,257 @@ description: Learn how to route traffic to different proxies using a service clu
 keywords: ucp, interlock, load balancing, routing
 ---
 
-## Configure Proxy Services
-With the node labels, you can re-configure the Interlock Proxy services to be constrained to the
-workers for each region. For example, from a manager, run the following commands to pin the proxy services to the ingress workers:
+Reconfiguring Interlock's proxy can take a long time (1-2 seconds) per overlay network managed by that proxy. In order to scale up to larger number of Docker networks and services routed to by Interlock, you may consider implementing *service clusters*: multiple proxy services managed by Interlock (rather than the default single proxy service), each responsible for routing to a separate set of Docker services and their corresponding networks, thereby minimizing proxy reconfiguration time.
```
Comment (adrian-plata) [inline ee/ucp/interlock/usage/service-clusters.md]: A couple things. The last sentence in this paragraph is a bit of a run on. Could you break it apart?

Also, stylistically we try to keep each line to 80 characters. Not a big issue, but it helps with reviewing stuff.

File: ee/ucp/interlock/usage/service-clusters.md [hunk 2]
```diff
@@ -4,205 +4,257 @@ description: Learn how to route traffic to different proxies using a service clu
...
+## Prerequisites
 
-```bash
-$> docker network create -d overlay demo-east
-$> docker network create -d overlay demo-west
-```
+In this example, we'll assume you have a UCP cluster set up with at least two worker nodes, `ucp-node-0` and `ucp-node-1`; we'll use these as dedicated proxy servers for two independent Interlock service clusters. 
 
-Add the networks to the Interlock configuration file. Interlock automatically adds networks to the proxy service upon the next proxy update. See *Minimizing the number of overlay networks* in [Interlock architecture](https://docs.docker.com/ee/ucp/interlock/architecture/) for more information. 
+We'll also assume you've already enabled Interlock, per the [instructions here](https://docs.docker.com/ee/ucp/interlock/deploy/), with an HTTP port of 80 and an HTTPS port of 8443.
```
Comment (adrian-plata) [inline ee/ucp/interlock/usage/service-clusters.md]: I believe the link here should be [instructions here](../../deploy/)

File: ee/ucp/interlock/usage/service-clusters.md [hunk 3]
```diff
@@ -4,205 +4,257 @@ description: Learn how to route traffic to different proxies using a service clu
...
 
-> Note
-> 
-> Interlock will _only_ connect to the specified networks, and will connect to them all at startup.
+## Setting Up Interlock Service Clusters
```
Comment (adrian-plata) [inline ee/ucp/interlock/usage/service-clusters.md]: Sentence case for titles.

##Setting up Interlock service clusters
(assuming Interlock is a proper noun)

File: ee/ucp/interlock/usage/service-clusters.md [hunk 4]
```diff
@@ -4,205 +4,257 @@ description: Learn how to route traffic to different proxies using a service clu
...
-$> docker node update --label-add nodetype=loadbalancer --label-add region=us-west lb-02
-lb-02
-$> docker node update --label-add nodetype=loadbalancer --label-add region=us-west lb-03
-lb-03
+CURRENT_CONFIG_NAME=$(docker service inspect --format '{{ (index .Spec.TaskTemplate.ContainerSpec.Configs 0).ConfigName }}' ucp-interlock)
+docker config inspect --format '{{ printf "%s" .Spec.Data }}' $CURRENT_CONFIG_NAME > old_config.toml
 ```
 
-Inspect each node to ensure the labels were successfully added:
+Make a new config file called `config.toml` with the following content, which declares two service clusters, `east` and `west`. **Note** you will have to change the UCP version (`3.2.3` in the example below) to match yours, as well as all instances of `*.ucp.InstanceID` (`vl5umu06ryluu66uzjcv5h1bo` below):
```
Comment (adrian-plata) [inline ee/ucp/interlock/usage/service-clusters.md]: Note should be stylized as:

> **Note:**
> You will have to change the UCP version (`3.2.3` in the
> example below) to match yours, as well as all instances
> of `*.ucp.InstanceID` (`vl5umu06ryluu66uzjcv5h1bo` below):

File: ee/ucp/interlock/usage/service-clusters.md [hunk 5]
```diff
@@ -4,205 +4,257 @@ description: Learn how to route traffic to different proxies using a service clu
...
+```bash
+docker service update \
+  --config-rm $CURRENT_CONFIG_NAME \
+  --config-add source=$NEW_CONFIG_NAME,target=/config.toml \
+  ucp-interlock
+```
+
+Finally, do a `docker service ls`. You should see two services providing Interlock proxies, `ucp-interlock-proxy-east` and `-west`. If you only see one Interlock proxy service, delete it with `docker service rm`. After a moment, the two new proxy services should be created, and Interlock will be successfully configured with two service clusters.
+
+## Deploying Services in Separate Service Clusters
```
Comment (adrian-plata) [inline ee/ucp/interlock/usage/service-clusters.md]: Sentence case

**Comment by BillMills** [comment]
Thanks for your comments, @adrian-plata, and sorry for the slow reply - all your comments implemented verbatim.

File: ee/ucp/interlock/usage/service-clusters.md [hunk 6]
```diff
@@ -4,205 +4,296 @@ description: Learn how to route traffic to different proxies using a service clu
...
-$> docker network create -d overlay demo-east
-$> docker network create -d overlay demo-west
-```
+In this example, we'll assume you have a UCP cluster set up with at least two 
+worker nodes, `ucp-node-0` and `ucp-node-1`; we'll use these as dedicated 
+proxy servers for two independent Interlock service clusters. 
 
-Add the networks to the Interlock configuration file. Interlock automatically adds networks to the proxy service upon the next proxy update. See *Minimizing the number of overlay networks* in [Interlock architecture](https://docs.docker.com/ee/ucp/interlock/architecture/) for more information. 
+We'll also assume you've already enabled Interlock, per the 
+[instructions here](../../deploy/),
```
Comment (adrian-plata) [inline ee/ucp/interlock/usage/service-clusters.md]: I broke the link. I think it should be `../deploy/` (only a single pair of .)

File: ee/ucp/interlock/usage/service-clusters.md [hunk 7]
```diff
@@ -4,205 +4,296 @@ description: Learn how to route traffic to different proxies using a service clu
...
 
-Add the networks to the Interlock configuration file. Interlock automatically adds networks to the proxy service upon the next proxy update. See *Minimizing the number of overlay networks* in [Interlock architecture](https://docs.docker.com/ee/ucp/interlock/architecture/) for more information. 
+We'll also assume you've already enabled Interlock, per the 
+[instructions here](../../deploy/), 
+with an HTTP port of 80 and an HTTPS port of 8443.
 
-> Note
-> 
-> Interlock will _only_ connect to the specified networks, and will connect to them all at startup.
+## Setting Up Interlock service clusters
```
Comment (adrian-plata) [inline ee/ucp/interlock/usage/service-clusters.md]: lowercase "up"

**Comment by BillMills** [comment]
@adrian-plata all your new comments implemented verbatim - sorry for the slow reply here, was on vacation all last week.

File: ee/ucp/interlock/usage/service-clusters.md [hunk 8]
```diff
@@ -4,205 +4,296 @@ description: Learn how to route traffic to different proxies using a service clu
 keywords: ucp, interlock, load balancing, routing
 ---
 
-## Configure Proxy Services
-With the node labels, you can re-configure the Interlock Proxy services to be constrained to the
-workers for each region. For example, from a manager, run the following commands to pin the proxy services to the ingress workers:
+Reconfiguring Interlock's proxy can take a long time (1-2 seconds) per overlay
```
Comment (traci-morrison) [inline ee/ucp/interlock/usage/service-clusters.md]: Suggest "...can take between 1-2 seconds per overlay...". The phrase "a long time" is ambiguous.

File: ee/ucp/interlock/usage/service-clusters.md [hunk 9]
```diff
@@ -4,205 +4,296 @@ description: Learn how to route traffic to different proxies using a service clu
...
+network managed by that proxy. In order to scale up to larger number of Docker
```
Comment (traci-morrison) [inline ee/ucp/interlock/usage/service-clusters.md]: s/In order to/To

**Comment by traci-morrison** [comment]
@dani-docker @euanh @trapier PTAL

**Comment by BillMills** [comment]
Thanks for the look @traci-morrison - your comments implemented verbatim in 6f5990c.

**Comment by traci-morrison** [comment]
> Thanks for the look @traci-morrison - your comments implemented verbatim in [6f5990c](https://github.com/docker/docker.github.io/commit/6f5990c47b3c22eaee7c6e54a53bd83dab36745a).

Thanks @BillMills. Is this ready to merge or do you have more edits?

**Comment by BillMills** [comment]
I'm good with this - did you receive a technical review from anyone you tagged? Would be good to get signoff from an expert in the feature, as this demo was mostly me muddling through.

**Comment by traci-morrison** [comment]
@BillMills The Infrastructure team is copying our repos over to Mirantis this week. Any open PRs and issues will be lost, so please do what you can to close or merge this PR before Thursday. Thank you!

**Comment by BillMills** [comment]
@traci-morrison ok, I resolved the merge conflict and kept the banner that was added to all the EE pages, but I don't have rights to merge this myself - it's up to you.

**Pull request merged**